### PR TITLE
Helm: Upgrade Minio chart dependency to 4.0.12.

### LINKED ISSUE #6973: [Helm] Update Minio chart version.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
Changes to a supported Minio chart. Minio is also enabled by default.

**Which issue(s) this PR fixes**:
Fixes #6973

**Checklist**
- [x] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: production/helm/loki/templates/NOTES.txt [hunk 1]
```diff
@@ -8,35 +8,16 @@ Installed components:
 {{- if .Values.gateway.enabled }}
 * gateway
 {{- end }}
+{{- if .Values.minio.enabled }}
+* minio 
+{{- end }}
 * read
 * write
 
+{{- if eq .Values.loki.storage.type "filesystem" }}
```
Comment (jeschkies) [inline production/helm/loki/templates/NOTES.txt:17]: Is this the correct way, @trevorwhitney?

File: production/helm/loki/values.yaml [hunk 2]
```diff
@@ -1062,7 +1062,7 @@ tracing:
 # Configuration for `minio` child chart
 # -------------------------------------
 minio:
-  enabled: false
+  enabled: true
```
Comment (jeschkies) [inline production/helm/loki/values.yaml:1065]: I think the Helm chart should be production ready.

File: tools/dev/k3d/jsonnetfile.lock.json [hunk 3]
```diff
@@ -1,24 +1,14 @@
 {
   "version": 1,
   "dependencies": [
-    {
```
Comment (jeschkies) [inline tools/dev/k3d/jsonnetfile.lock.json]: This came from jb. I should revert it.

File: tools/dev/k3d/Makefile [hunk 4]
```diff
@@ -44,5 +44,5 @@ prepare: create-registry create-cluster update-repos build-latest-image
 
 build-latest-image:
 	make -C $(CURDIR)/../../.. loki-image
-	docker tag grafana/loki:$(IMAGE_TAG) k3d-grafana:$(REGISTRY_PORT)/loki:latest
-	docker push k3d-grafana:$(REGISTRY_PORT)/loki:latest
+	docker tag grafana/loki:$(IMAGE_TAG) grafana.k3d.localhost:$(REGISTRY_PORT)/loki:latest
+	docker push grafana.k3d.localhost:$(REGISTRY_PORT)/loki:latest
```
Comment (jeschkies) [inline tools/dev/k3d/Makefile:48]: This is a funny thing. For some reason `grafana.k3d.localhost` works. It also does on `grafana/cloud-onboarding`.

File: production/helm/loki/values.yaml (hunk 2 above)
Comment (trevorwhitney) [reply production/helm/loki/values.yaml:1065]: I am ok with this change, though we need to make sure to change the necessary `values.yaml` in the upgrade guide in the README for people coming from `grafana/loki` who will not have had a minio.

```
minio:
  enabled: false
```

Comment (trevorwhitney) [reply production/helm/loki/values.yaml:1065]: Hmm, actually, on second thought, we do not recommend minio for production. so I think having it off by default might be more "production ready" as it's assuming they're providing s3/gcs/azure credentials. WDYT?

File: tools/dev/k3d/Makefile (hunk 4 above)
Comment (trevorwhitney) [reply tools/dev/k3d/Makefile:48]: yup, works for me too :+1:

**Review by trevorwhitney — CHANGES_REQUESTED** [review]
Looks great, thanks for adding this. Curious about your feedback to my thoughts on enabling minio by default. I also think we should just nuke that whole section in `NOTES.txt` about filesystem backend not working.

File: production/helm/loki/templates/NOTES.txt (hunk 1 above)
Comment (jeschkies) [reply production/helm/loki/templates/NOTES.txt:17]: Should we drop this entry for now and just link the documentation once we have them?

File: production/helm/loki/values.yaml (hunk 2 above)
Comment (jeschkies) [reply production/helm/loki/values.yaml:1065]: Oh, I didn't know we don't recommend it. Ok, I'll change this back.

File: production/helm/loki/values.yaml [hunk 5]
```diff
@@ -1059,7 +1059,7 @@ tracing:
 # Configuration for `minio` child chart
 # -------------------------------------
 minio:
-  enabled: true
+  enabled: false
```
Comment (trevorwhitney) [inline production/helm/loki/values.yaml:1062]: this looks like a bug, do we have the `minio` section twice? I think we need to merge this `enabled` into the section below it

**Review by trevorwhitney — APPROVED** [review]
LGTM. One small item about the `values.yaml` to fix before merge, but this looks great. Thanks again!

File: production/helm/loki/values.yaml (hunk 5 above)
Comment (trevorwhitney) [reply production/helm/loki/values.yaml:1062]: wait, nevermind, the diff was doing something weird for me, this is good

**Review by trevorwhitney — APPROVED** [review]
Nevermind my previous comment, the diff was doing something weird for me, but when I looked at the actual file it loos fine. LGTM!

**Pull request merged**